eras/shelley/impl/src/Cardano/Ledger/Shelley/Genesis.hs

```diff
@@ -129,6 +149,71 @@ emptyGenesisStaking =
       sgsStake = mempty
     }
 
+-- | Unlike @'NominalDiffTime'@ that supports @'Pico'@ precision, this type
+-- only supports @'Micro'@ precision.
+newtype NominalDiffTimeMicro = NominalDiffTimeMicro Micro
+  deriving (Show, Eq, Generic)
+  deriving anyclass (NoThunks)
+  deriving newtype (Ord, Num, Fractional, Real, ToJSON, FromJSON)
+
+deriving instance Generic Micro
+
+deriving anyclass instance NoThunks Micro
+
+instance ToCBOR NominalDiffTimeMicro where
+  toCBOR = encodeNominalDiffTimeMicro
+
+encodeNominalDiffTimeMicro :: NominalDiffTimeMicro -> Encoding
+encodeNominalDiffTimeMicro (NominalDiffTimeMicro micro) =
+  encodeInteger . toMicroseconds $ micro
+  where
+    toMicroseconds t =
+      numerator (toRational t * toRational (resolution $ Proxy @E6))
```

Comment:
This one is much easier ;) 
```suggestion
encodeNominalDiffTimeMicro (NominalDiffTimeMicro (MkFixed micro)) = encodeInteger micro
```

Reply:
Now that we have the instance for `Micro` the codec was easy to convert to `deriving newtype (ToCBOR, FromCBOR)`